src/dev/amdgpu/pm4_packet_processor.cc

```diff
@@ -664,7 +664,37 @@ PM4PacketProcessor::unmapQueues(PM4Queue *q, PM4UnmapQueues *pkt)
                 delete mqd;
             }
         }
-        gpuDevice->deallocateAllQueues();
+        gpuDevice->deallocateAllQueues(true);
+        } break;
+      case 3: {
+        auto &hsa_pp = gpuDevice->CP()->hsaPacketProc();
+        for (auto iter : gpuDevice->getUsedVMIDs()) {
+            for (auto id : iter.second) {
+                assert(queues.count(id));
+
+                // Do not unmap KMD queues or static queues
+                if (queues[id]->privileged() || queues[id]->isStatic()) {
+                    continue;
+                }
+                QueueDesc *mqd = queues[id]->getMQD();
+                DPRINTF(PM4PacketProcessor, "Unmapping queue %d with read "
+                        "index %ld\n", id, mqd->mqdReadIndex);
+                // Partially writing the mqd with an offset of 96 dwords
```

Comment:
I know this isn't immediately from this commit, but it would help if we explained why 96 dwords

Reply:
done